### PR TITLE
[dynamo][yolov3] Track UnspecializedNNModuleVariable for mutation

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128590
* __->__ #128269
* #128715

Fixes https://github.com/pytorch/pytorch/issues/101168

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Review by jansel — APPROVED** [review]

**Comment by tombousso** [comment]
@anijain2305 this PR introduces a bug because the side effects object gets created multiple times, overwriting the side effects.

To reproduce:

```
models = nn.Sequential(nn.Linear(3, 3))
models[0].abc = False

def run():
    models[0].abc = True
    x = torch.randn(1, 3)
    return models(x)

run = torch.compile(run, fullgraph=True)
run()
print(models[0].abc) # this prints False instead of True!
```

One fix I found is adding a cache: https://github.com/pytorch/pytorch/blob/7775fee10f31ee683bd7beee9a5a9829c6574637/torch/_dynamo/side_effects.py#L214-L221

to:

```
    def _track_obj(
        self,
        item: Any, 
        variable: VariableTracker,
        mutable_cls=MutableSideEffects,
    ):
        """Start tracking a new variable for mutation"""
        assert variable.source is not None 
        if id(item) in self.id_to_variable:
            variable.mutable_local = self.id_to_variable[id(item)].mutable_local
            return variable
```

File: benchmarks/dynamo/ci_expected_accuracy/aot_eager_torchbench_training.csv [hunk 1]
```diff
@@ -286,4 +286,4 @@ vision_maskrcnn,pass,34
 
 
 
-yolov3,pass,9
+yolov3,fail_accuracy,9
```
Comment (anijain2305) [inline benchmarks/dynamo/ci_expected_accuracy/aot_eager_torchbench_training.csv:289]: This seems to be flaky. I went into the SSH machine.

**Pull request closed without merging**